HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+        if (/.*1$/.test(totalEmployees) && !/.*11$/.test(totalEmployees)) {
+            console.log(`${enterprise.name} (${totalEmployees} сотрудник)`);
+        } else {
+        console.log(`${enterprise.name} (${totalEmployees} сотрудников)`);
+        }
+        enterprise.departments.map(department => {
+            if (/.*1$/.test(department.employees_count) && !/.*11$/.test(department.employees_count)) {
+                console.log(`- ${department.name} (${department.employees_count} сотрудник)`);
+            } else {
+            console.log(`- ${department.name} (${department.employees_count} сотрудников)`);
+            }
+        });
+    });
+}
+
+enterpriseStructure(enterprises);
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+// Функция-помощник для получения предприятия по id отдела или названию отдела:
+const getEnterpriseByDepartment = (idOrName) => enterprises.filter((enterprise) => enterprise.departments.find((department) => department.id === idOrName || department.name === idOrName));
+
+const getEnterpriseName = (idOrName) => {
+  const enterpriseFound = getEnterpriseByDepartment(idOrName);
+  if (enterpriseFound == 0) throw new Error('Нет предприятия с данным id отдела или названием отдела');
+  
+  return enterpriseFound.map(enterprise => enterprise.name).join(', ');
+}
+
+console.log(getEnterpriseName(4));
+console.log(getEnterpriseName("Отдел маркетинга"));
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+// Функция-помощник для создания нового id
+const newId = () => {
+  ids = [];
+  enterprises.forEach((enterprise) => {
+    ids.push(enterprise.id);
+    ids.push(enterprise.departments.map(department => department.id));
+  });
+  const maxId = Math.max(...ids.flat());
+  
+  return maxId + 1;
+};
+
+const addEnterprise = (newEnterprise) => {
+  enterprises.push({
+    id: newId(enterprises),
+    name: newEnterprise,
+    departments: [],
+  })
+  
+  return JSON.stringify(enterprises);
+}
+
+console.log(addEnterprise('Предприятие 4'));
+
+// 4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+
+// Пример:
+// addDepartment(1, "Название нового отдела")
+
+// Функция-помощник для получения предприятия по id предприятия
+const getEnterpriseById = (id, []) =>
+enterprises.filter((enterprise) => enterprise.id === id);
+
+const addDepartment = (id, newDepartment) => {
+  const enterpriseFound = getEnterpriseById(id, enterprises);
+  if (enterpriseFound != 0) {
+    enterpriseFound[0].departments.push({
+      id: newId(enterprises),
```

Comment:
```suggestion
      id: newId(),
```
newId function does not have arguments

Reply:
Thanks, I realized the mistake